src/main/java/org/jabref/gui/EntryTypeView.java

```diff
@@ -0,0 +1,256 @@
+                        //If there are duplicates starts ImportInspectionDialog
+                        final BasePanel panel = basePanel;
+
+                        ImportInspectionDialog diag = new ImportInspectionDialog(basePanel.frame(), panel, Localization.lang("Import"), false);
+                        diag.addEntries(Arrays.asList(bibEntry));
+                        diag.entryListComplete();
+                        diag.setVisible(true);
+                        diag.toFront();
+                    } else {
+                        // Regenerate CiteKey of imported BibEntry
+                        new BibtexKeyGenerator(basePanel.getBibDatabaseContext(), prefs.getBibtexKeyPatternPreferences()).generateAndSetKey(bibEntry);
+                        // Update Timestamps
+                        if (prefs.getTimestampPreferences().includeCreatedTimestamp()) {
+                            bibEntry.setField(prefs.getTimestampPreferences().getTimestampField(), prefs.getTimestampPreferences().now());
+                        }
+                        basePanel.insertEntry(bibEntry);
+                    }
+
+                    close();
+                } else if (searchID.trim().isEmpty()) {
+                    dialogService.showWarningDialogAndWait(Localization.lang("Empty search ID"), Localization.lang("The given search ID was empty."));
+                } else if (!fetcherException) {
+                    dialogService.showErrorDialogAndWait(Localization.lang("No files found.", Localization.lang("Fetcher '%0' did not find an entry for id '%1'.", fetcher.getName(), searchID) + "\n" + fetcherExceptionMessage));
+                } else {
+                    dialogService.showErrorDialogAndWait(Localization.lang("Error"), Localization.lang("Error while fetching from %0", fetcher.getName()) + "." + "\n" + fetcherExceptionMessage);
+                }
+                fetcherWorker = new FetcherWorker();
+
+                idTextField.requestFocus();
+                idTextField.selectAll();
+                viewModel.searchingProperty().setValue(false);
+
+            } catch (ExecutionException | InterruptedException e) {
+                LOGGER.error(String.format("Exception during fetching when using fetcher '%s' with entry id '%s'.", searchID, fetcher.getName()), e);
+            }
+        }
+
+        @Override
+        protected Optional<BibEntry> call() throws Exception {
+            Optional<BibEntry> bibEntry = Optional.empty();
+
+            viewModel.searchingProperty().setValue(true);
+            viewModel.storeSelectedFetcher();
+            fetcher = viewModel.selectedItemProperty().getValue();
+            searchID = idTextField.getText();
+            if (!searchID.isEmpty()) {
+                try {
+                    bibEntry = fetcher.performSearchById(searchID);
+                } catch (FetcherException e) {
```

Comment:
There is no need to catch exceptions here. The Task class has already ways built-in to handle exceptions and signal errors (you need to overwrite a different method to handle exceptions). It might be easier to just use the `BackgroundTask` wrapper.